drivers/include/isl29125.h

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2015 Ludwig Ortmann
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    driver_isl29125 ISL29125 light sensor
+ * @ingroup     drivers
+ * @brief       Device driver for the ISL29125 RGB light sensor
+ *
+ * ## Description
+ *
+ * The device supports level conversion in 12, and 16 bit depth per
+ * channel. Selecting a higher precision results in a longer
+ * conversion time.
+ *
+ * ## Usage
+ *
+ * Examine `tests/driver_isr29125` for an exemplary application using
+ * this driver.
+ *
+ * ## Caveats
+ *
+ * A GPIO pin definition has been included in the interface for future
+ * compatibility only.
+ * - The driver does not support SYNC mode at the moment.
+ * - The driver supports program driven reading only, i.e. interrupt
+ *   mode is currently not supported.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for the ISL29125 RGB light sensor
+ *
+ * @author      Ludwig Ortmann <ludwig.ortmann@fu-berlin.de
+ */
+
+#ifndef ISL29125_H
+#define ISL29125_H
+
+#include <stdint.h>
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+ /**
+  * @brief The sensors hard coded I2C address
+  */
+#define ISL29125_I2C_ADDRESS 0x44
+
+/**
+ * @brief data type for storing rgb sensor readings
+ */
+typedef struct {
+    float red;              /**< red lux value */
+    float green;            /**< green lux value */
+    float blue;             /**< blue lux value */
+} isl29125_rgb_t;
+
+/**
+ * @brief supported operation modes of the ISL29125 sensor's ADC
+ * conversion
+ */
+typedef enum {
+    ISL29125_MODE_DOWN = 0x00,      /**< ADC conversion powered down */
+    ISL29125_MODE_STANDBY = 0x04,   /**< ADC conversion not performed */
+    ISL29125_MODE_RGB = 0x05,       /**< RGB conversion */
+    ISL29125_MODE_R = 0x02,         /**< red conversion only */
+    ISL29125_MODE_G = 0x01,         /**< green conversion only */
+    ISL29125_MODE_B = 0x03,         /**< blue conversion only */
+    ISL29125_MODE_RG = 0x06,        /**< red and green conversion only */
+    ISL29125_MODE_GB = 0x07         /**< green and blue conversion only */
+} isl29125_mode_t;
```

Comment:
No, this was because of https://github.com/RIOT-OS/RIOT/pull/2906#discussion-diff-29600762